src/orders/components/OrderRefundDatagrid/OrderRefundDatagrid.tsx

```diff
@@ -0,0 +1,144 @@
+import { DashboardCard } from "@dashboard/components/Card";
+import { ColumnPicker } from "@dashboard/components/Datagrid/ColumnPicker/ColumnPicker";
+import { useColumns } from "@dashboard/components/Datagrid/ColumnPicker/useColumns";
+import Datagrid from "@dashboard/components/Datagrid/Datagrid";
+import {
+  DatagridChangeStateContext,
+  useDatagridChangeState,
+} from "@dashboard/components/Datagrid/hooks/useDatagridChange";
+import { useEmptyColumn } from "@dashboard/components/Datagrid/hooks/useEmptyColumn";
+import { OrderDetailsFragment } from "@dashboard/graphql";
+import useListSettings from "@dashboard/hooks/useListSettings";
+import { orderGrantRefundEditUrl } from "@dashboard/orders/urls";
+import { ListViews } from "@dashboard/types";
+import {
+  Box,
+  Button,
+  EditIcon,
+  PlusIcon,
+  Text,
+  useTheme,
+} from "@saleor/macaw-ui-next";
+import React from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+import { Link } from "react-router-dom";
+
+import {
+  createGetCellContent,
+  orderRefundStaticColumnsAdapter,
+} from "./datagrid";
+import { refundGridMessages } from "./messages";
+
+interface OrderRefundDatagridProps {
+  grantedRefunds: OrderDetailsFragment["grantedRefunds"];
+  orderId: string;
+}
+
+export const OrderRefundDatagrid: React.FC<OrderRefundDatagridProps> = ({
```

Comment:
This component mixes two contexts: grid operations and refunds rendering, it can be separated:
- hooks related to this specific grid can be grouped into one (or more?)
- hooks related to refunds itself can be... also delegated (if it makes sense)